### PR TITLE
Support ImplicitUsings in project and item templates

### PR DESCRIPTION
**Pull request by DavidBoike**

### CODE REVIEW

File: src/ParticularTemplates/Saga/.template.config/template.json [hunk 1]
```diff
@@ -20,6 +20,11 @@
       "replaces": "NamespaceName",
       "defaultValue": "Handlers"
     },
+    "ImplicitUsings": {
+      "type": "bind",
+      "binding": "msbuild:ImplicitUsings",
+      "replaces": "ImplicitUsingsValue"
```
Comment (bording) [inline src/ParticularTemplates/Saga/.template.config/template.json:26]: Is this needed? The other item template doesn't have this line and I don't see where it's being used.

File: src/Tests/TemplateTests.cs [hunk 2]
```diff
@@ -111,7 +111,10 @@ public async Task NServiceBusEndpointPersistence(string persistence)
     }
 
     [Test]
-    public async Task Handler()
+    [TestCase("Default")]
+    // Tests for ImplicitUsings not available
+    [TestCase("net48")]
```
Comment (bording) [inline src/Tests/TemplateTests.cs:116]: Given that that `ImplicitUsings` condition is : `"value": "(framework != \"net472\" && framework != \"net48\")"`, should we be testing more cases here?

**Comment by bording** [comment]
Given that I think we should also be including `<Nullable>enable</Nullable>`, I'm wondering if we should instead have a `IsNetFramework` property that could be used to handle `Nullable` and `ImplicitUsings`.

File: src/Tests/TemplateTests.cs (hunk 2 above)
Comment (DavidBoike) [reply src/Tests/TemplateTests.cs:116]: I only wanted to test the two conditions. That the frameworks chosen result in different ImplicitUsings outcomes are verified by other tests.

**Review by bording — APPROVED** [review]

**Review by andreasohlund — APPROVED** [review]

**Pull request merged**